uiowa/uiowa
# Add video functionality to banner block

Resolves https://github.com/uiowa/uiowa/issues/2403. 

## How to test video and autoplay checkbox
1. `composer install`
2. `vagrant up --provision`
3.  `blt ds --site=admissions.uiowa.edu`
4. `blt frontend`
5. Add a video to an existing banner block, and verify that the video is loaded in paused state with a preview poster image both on the layout builder preview and saved page. 
6. Verify that the "autoplay" checkbox only appears if a video has been selected from the media library. 
7. Edit the banner block and check the "autoplay" checkbox and verify that video autoplays in layout builder preview and once the page is save. 
8. Edit the block again and uncheck the "autoplay" box and verify that autoplay stops in layout builder preview and saved page. 
9. Open the page in a new browser window and verify that the video loads based on your settings. 
10. Add another video banner to the same page and verify that you can set the autoplay option for the new banner block. 
11. Change your system accessibility settings to "Reduce motion" and verify that `prefers-reduced-motion` stops any video from autoplay. 

## How to test video cookie
(If at any point you would like to view the video uuids that are paused or how many videos the site is tracking for pausing, you can put `JSON.parse(jQuery.cookie('uiowa-video'))` in your console)
1. pause any video used for a banner in a published page.
2. Upon page reload, you should see that the video is still paused. You should also be able to move away from the page and come back and observe the same behavior. (If you were really die hard about testing this cookie, you could wait the 99 years that it will last to make sure it expires correctly)
3. create more video banners on the same page and test that they all have play and pause cookies that function independently.
4. Create a new page that also contains any number of video banners. 
5. Check that you can also set cookies for pausing on this page as well.
6. make sure that you can navigate away and between your video banner pages, and that the cookies persist through page closing, offsite navigation, etc.
